packages/clerk-js/src/ui/components/OrganizationProfile/MembersSearch.tsx

```diff
@@ -0,0 +1,113 @@
+import type { useOrganization } from '@clerk/shared/react';
+import type { GetMembersParams } from '@clerk/types';
+import { useEffect, useRef } from 'react';
+
+import { Flex, localizationKeys, useLocalizations } from '../../../ui/customizables';
+import { Animated, InputWithIcon } from '../../../ui/elements';
+import { MagnifyingGlass } from '../../../ui/icons';
+import { Spinner } from '../../../ui/primitives';
+import { ACTIVE_MEMBERS_PAGE_SIZE } from './OrganizationMembers';
+
+type MembersSearchProps = {
+  /**
+   * Controlled query param state by parent component
+   */
+  query: GetMembersParams['query'];
+  /**
+   * Controlled input field value by parent component
+   */
+  value: string;
+  /**
+   * Paginated organization memberships
+   */
+  memberships: ReturnType<typeof useOrganization>['memberships'];
+  /**
+   * Handler for change event on input field
+   */
+  onSearchChange: (value: string) => void;
+  /**
+   * Handler for `query` value changes
+   */
+  onQueryTrigger: (query: string) => void;
+};
+
+const membersSearchDebounceMs = 500;
+
+export const MembersSearch = ({ query, value, memberships, onSearchChange, onQueryTrigger }: MembersSearchProps) => {
+  const { t } = useLocalizations();
+
+  const inputRef = useRef<HTMLInputElement | null>(null);
+  const debounceTimer = useRef<NodeJS.Timeout | undefined>();
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const value = event.target.value;
```

Comment:
Looks like this is shadowing `value` from the outer scope. 

```suggestion
    const eventValue = event.target.value;
```

Reply:
Fixed on https://github.com/clerk/javascript/pull/4942/commits/6b7918cd24f8163a00ecc118f541fc1d265c010f